### PR TITLE
Add DMC standalone

### LINKED ISSUE #1698: Duff moisture code
### Addressing a Problem?

I would like to use the Duff moisture code for wildfire analysis

### Potential Solution

Similar to https://xclim.readthedocs.io/en/stable/indices.html#xclim.indices.drought_code

### Additional context

See https://wikifire.wsl.ch/tiki-index9436.html?page=Duff+moisture+code

### Contribution

- [ ] I would be willing/able to open a Pull Request to contribute this feature.

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### PR DESCRIPTION
**Pull request by aulemahal**
### Pull Request Checklist:
- [x] This PR addresses an already opened issue (for bug fixes / features)
    - This PR fixes #1698
- [x] Tests for the changes have been added (for bug fixes / features)
  - [x] (If applicable) Documentation has been added / updated (for bug fixes / features)
- [x] CHANGES.rst has been updated (with summary of main changes)
  - [x] Link to issue (:issue:`number`) and pull request (:pull:`number`) has been added

### What kind of change does this PR introduce?

* Adds an indicator entry point for computing the duff moisture code (DMC) alone. Alike what is already done for the drought code (DC).
* I added tests for explicitly testing the two standalone indicators.
* Minor adjustment for a better error message when passing unneeded parameters to DC and DMC.

### Does this PR introduce a breaking change?
No.

### Other information:
FYI @raybellwaves

### CODE REVIEW

**Comment by raybellwaves** [comment]
Thanks!

File: xclim/indicators/atmos/_precip.py [hunk 1]
```diff
@@ -394,7 +395,6 @@ class HrPrecip(Hourly):
     title="Daily drought code",
     identifier="dc",
     units="",
-    standard_name="drought_code",
```
Comment (Zeitsperre) [inline xclim/indicators/atmos/_precip.py:397]: I'm guessing that it's not in the CF Conventions?

File: xclim/indices/fire/_cffwis.py [hunk 2]
```diff
@@ -1247,12 +1248,13 @@ def overwintering_drought_code(
...
-# TODO: The `noqa` can be removed when this function is no longer public
-def _convert_parameters(params: dict[str, int | float]) -> dict[str, int | float]:
+def _convert_parameters(
+    params: dict[str, int | float], funcname: str = "fire weather indices"
+) -> dict[str, int | float]:
     for param, value in params.copy().items():
         if param not in default_params:
             raise ValueError(
-                f"{param} is not a valid parameter for fire weather indices. See list in xc.indices.fire.default_params."
+                f"{param} is not a valid parameter for {funcname}. See the docstring of the function and the list in xc.indices.fire.default_params."
```
Comment (Zeitsperre) [inline xclim/indices/fire/_cffwis.py:1257]: Nicely done.

**Review by Zeitsperre — APPROVED** [review]
Elegant, clean.

“What is once well done is done forever.” - Henry David Thoreau

File: xclim/data/fr.json [hunk 3]
```diff
@@ -799,6 +799,12 @@
     "long_name": "Indice forêt météo",
     "description": "Évaluation numérique de l'intensité du feu."
   },
+  "DMC": {
```
Comment (Zeitsperre) [inline xclim/data/fr.json:802]: @aulemahal I leave the verification here to you.

File: xclim/data/fr.json [hunk 4]
```diff
@@ -799,6 +799,12 @@
...
+    "long_name": "Indice d'humidité de l'humus (DMC)",
+    "description": "Évaluation numérique de la teneur en eau moyenne des couches organiques peu compactes d'une profondeur modérée.",
+    "title": "Indice d'humidité de l'humus",
+    "abstract": "Évaluation numérique de la teneur en eau moyenne des couches organiques peu compactes d'une profondeur modérée."
```
Comment (aulemahal) [inline xclim/data/fr.json:806]:
```suggestion
    "long_name": "Indice de l'humus",
    "description": "Évaluation numérique de la teneur moyenne en eau des couches organiques peu tassées de moyenne épaisseur.",
    "title": "Indice de l'humus (IH)",
    "abstract": "Évaluation numérique de la teneur moyenne en eau des couches organiques peu tassées de moyenne épaisseur."
```
This is what the FWI indicator puts for DMC, and it comes from an official document. I liked yours better though...
And I just realize we don't have the correct long name for the FWI one :roll_eyes: .

https://cwfis.cfs.nrcan.gc.ca/renseignements/sommaire/fwi

**Pull request merged**